gateway/ln-gateway/src/client.rs

```diff
@@ -48,21 +47,31 @@ impl StandardGatewayClientBuilder {
         tg: TaskGroup,
         old_client: Option<fedimint_client::Client>,
     ) -> Result<fedimint_client::Client> {
-        let federation_id = config.config.federation_id;
+        let FederationConfig {
+            connection,
+            mint_channel_id,
+            timelock_delta,
+            fees,
+        } = config;
+        let federation_id = connection.id;
+
+        // TODO: remove after <https://github.com/fedimint/fedimint/issues/2506>
+        let api: DynGlobalApi = WsFederationApi::from_connect_info(&[connection.clone()]).into();
+        let client_config = api.download_client_config(&connection).await?;
```

Comment:
If this fails, will `gatewayd` fail to start, or does it just skip initializing this client for the federation?

Reply:
Skips initializing client for the federation. So a `/connect-fed` request fails

Reply:
I believe these clients are initialized when the gateway first boots up though too. So if federation X is offline when `gatewayd` boots up, does that prevent `gatewayd` from starting correctly?

Reply:
I think ideally if this fails we should just catch the failure and emit a warning message or something that we failed to load the client for that federation (and at some point make that visible in the UI)

Reply:
> I believe these clients are initialized when the gateway first boots up though too. So if federation X is offline when gatewayd boots up, does that prevent gatewayd from starting correctly

I hadn't considered the case of re-initializing clients on gatewayd boot. It's true failure to load any single client results in failure to load gateway.

With this change where we always download config from federation, we have higher chances of failing to load gateway!

> I think ideally if this fails we should just catch the failure and emit a warning message or something that we failed to load the client for that federation (and at some point make that visible in the UI)

Agreed. Tracking here https://github.com/fedimint/fedimint/issues/2857